File: 2023/advent_of_code/test/q9/q9_test.exs
defmodule AdventOfCode.Q9Test do
  use ExUnit.Case
  doctest AdventOfCode.Q9

  test "Q9 Part 1 Example works" do
    file_stream = File.stream!("./test/q9/q9_example.txt")
    result = AdventOfCode.Q9.part1(file_stream)
    assert result == 114
  end

  test "Q9 Part 1 works" do
    file_stream = File.stream!("./test/q9/q9.txt")
    result = AdventOfCode.Q9.part1(file_stream)
    assert result == 1_901_217_887
  end

  test "Q9 Part 2 Example works" do
    file_stream = File.stream!("./test/q9/q9_example.txt")
    result = AdventOfCode.Q9.part2(file_stream)
    assert result == 2
  end

  test "Q9 Part 2 works" do
    file_stream = File.stream!("./test/q9/q9.txt")
    result = AdventOfCode.Q9.part2(file_stream)
    assert result == 905
  end
end

File: 2022/AdventOfCode/src/test/kotlin/q9/Q9Test.kt
package q9

import java.nio.file.Paths
import kotlin.test.Test
import kotlin.test.assertEquals

class Q9Test {
    @Test
    fun `part 1 works`() {
        val filePath = javaClass.getResource("/q9.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(6175, part1(filePath!!))
    }

    @Test
    fun `part 2 works`() {
        val filePath = javaClass.getResource("/q9.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(2578, part2(filePath!!))
    }
}